src/HeapOrdering.sol

```diff
@@ -248,19 +234,23 @@ library HeapOrdering {
         uint96 _removedValue
     ) private {
         uint256 index = _heap.indexOf[_id];
+        delete _heap.indexOf[_id];
         uint256 accountsLength = _heap.accounts.length;
 
-        // Swap the last account and the account to remove, then pop it.
-        swap(_heap, index, accountsLength - 1);
         if (_size == accountsLength) _heap.size = --_size;
+        if (index == accountsLength - 1) {
+            _heap.accounts.pop();
+            return;
+        }
+
+        Account memory lastAccount = _heap.accounts[accountsLength - 1];
         _heap.accounts.pop();
-        delete _heap.indexOf[_id];
 
         // If the swapped account is in the heap, restore the invariant: its value can be smaller or larger than the removed value.
```

Comment:
do we even keep this line? because it does not appear in `ThreeHeapOrdering.sol`, and to objective is to harmonize both.

- either we correct it in `HeapOrdering` and add it in `ThreeHeapOrdering.sol`
- eiter we delete it from  `HeapOrdering`

Reply:
Both are ok with me, I'll let you decide

Reply:
I kept the line